### PR TITLE
Upgrade to substrait 0.5.1 and set the version field of produced plans

### LINKED ISSUE #4949: Write correct value to `version` field when serializing plans to Substrait
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
See https://github.com/apache/arrow-datafusion/pull/4895/files#r1069790686

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mbrobbel**
# Which issue does this PR close?

Closes #4949.

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs
Comment (waynexia) [inline datafusion/substrait/src/logical_plan/producer.rs:1]: Those import style changes are not necessary, do you think we need to add some rustfmt rules (like [those](https://github.com/GreptimeTeam/greptimedb/blob/develop/rustfmt.toml) in greptimedb)? cc @andygrove @alamb

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -80,7 +83,7 @@ pub fn to_substrait_plan(plan: &LogicalPlan) -> Result<Box<Plan>> {
 
     // Return parsed plan
     Ok(Box::new(Plan {
-        version: None, // TODO: https://github.com/apache/arrow-datafusion/issues/4949
+        version: Some(version::version_with_producer("datafusion")),
```
Comment (waynexia) [inline datafusion/substrait/src/logical_plan/producer.rs:86]: We have both logical plan and physical plan. Is it needed to distinguish them in this version field?

File: datafusion/substrait/src/logical_plan/producer.rs
Comment (mbrobbel) [reply datafusion/substrait/src/logical_plan/producer.rs:1]: This is what `rustfmt` made of it after adding a use statement for the `version` mod.

I would not recommend using unstable configuration options for `rustfmt`.

File: datafusion/substrait/src/logical_plan/producer.rs (hunk 1 above)
Comment (mbrobbel) [reply datafusion/substrait/src/logical_plan/producer.rs:86]: No, that is not something that should be encoded in the version field of a plan.

**Review by andygrove — APPROVED** [review]

**Pull request merged**